stan/math/prim/mat/prob/categorical_logit_glm_lpmf.hpp

```diff
@@ -64,7 +64,7 @@ categorical_logit_glm_lpmf(
   check_bounded(function, "categorical outcome out of support", y, 1,
                 N_classes);
 
-  if (size_zero(y, x, beta)) {
+  if (size_zero(y, x, beta) || N_classes == 1) {
```

Comment:
Is thus a bug in the CPU version you are addressing here or just a performance thing for this case?

Reply:
A bit of both and a bit of simplifying testing. The problem happend with testing this case. Result was correctly calculated as (very close to) 0 in both CPU and GPU impleemntation. However, there were a bit different numerical errors (absolutely tiny, but relatively large) that caused `expect_near_rel()` to fail the test.